test/functional/wallet_migration.py

```diff
@@ -1018,6 +1018,12 @@ def test_migrate_simple_watch_only(self):
         res, _ = self.migrate_and_get_rpc("bare_p2pk")
         wo_wallet = self.master_node.get_wallet_rpc(res['watchonly_name'])
         assert_equal(wo_wallet.listdescriptors()['descriptors'][0]['desc'], descsum_create(f'pk({pubkey.hex()})'))
+
+        # Ensure that migrating a wallet with watch-only scripts does not create a spendable wallet.
+        assert_equal('bare_p2pk_watchonly', res['wallet_name'])
+        assert "bare_p2pk" not in self.master_node.listwallets()
+        assert "bare_p2pk" not in [w["name"] for w in self.master_node.listwalletdir()["wallets"]]
```

Comment:
Should also check that the backup file is still present.

Reply:
Done. Thx.